File: Two Pointers/containerWithWater.java
class Solution {
    public int maxArea(int[] height) {
        if(height.length <2) return 0;
        int left =0, right = height.length -1, ans = Integer.MIN_VALUE;
        while(left<right){
            ans = Math.max(ans, (right-left)*Math.min(height[left],height[right]));
            if(height[left]>height[right]) right--;
            else left++;
        }
        return ans;
    }
}

File: Two Pointers/containerWithWater.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        if len(height) < 2:
            return 0
        
        l,r,ans = 0, len(height)-1, 0
        while l<r:
            ans = max(ans, (r-l)*min(height[l],height[r]))
            if height[l]>height[r]:
                r-=1
            else:
                l+=1
        return ans
        